### PR TITLE
Alan fix subtask folder fix

### PR DESCRIPTION
**Pull request by LeeSingChanYau**
**This PR fixes WBS Subtask ‘folder’ button and edit bar showing incorrectly**
When having a sub task folder open, close and reopen the mother task’s folder. The sub task folder will remain open but not showing any sub tasks
Also when having a sub task edit bar open, close the mother task’s folder. The edit bar for that subtask won’t disappear
Example on the [video](https://drive.google.com/file/d/1USQPxQk-XefglMeRjmiwMqzycbodkjss/view?usp=sharing)
Here is the video of the problem as well.

https://user-images.githubusercontent.com/67836976/224519149-1aa5c5c6-6562-45e6-9dcd-1c2ba516b302.mp4

How to test:
- Log in as admin.
- Go to projects.
- Choose a project ( I chose AAA USE THIS PROJECT / Jae's WBS - USE THIS ).
- Open a folder, then open a subfolder that has subtasks or subfolders.
- Close the first folder you opened, and then reopen.
- Check if all subfolders are closed.
- Now open another folder.
- Click the Edit button for one of subtasks or subfolders.
- Close the first folder you opened, and check if edit options are gone.
- Reopen the folder and check that you cannot see the edit options.

Please check the video to understand the problem better.

### CODE REVIEW

**Review by NataliaCavicchioli — COMMENTED** [review]
Hi @LeeSingChanYau, I reviewed your PR and it is looking good. Good job!

I just noticed that when I open a subfolder and open the Edit, and close the first folder, the Edit button doesn't close. 

So, if I open the first folder's button, both of them remain open. Print below how it looks.

image

**Review by nicollext — COMMENTED** [review]
Hey @LeeSingChanYau I tested the PR and found some issues. 

For the Food folder the logic works well, but I need to click twice to open the first sub folder.


https://user-images.githubusercontent.com/101591708/225422840-8f04a9db-7d71-4a04-8d1a-525b2a6f45ee.MP4


For other folders when I close the main folder, the subtask remains on the page, and the edit tab does not disappear. This issue also occurred when I created a new folder to test.

https://user-images.githubusercontent.com/101591708/225422897-38bf55cc-fef4-4d91-b662-58a9e05d4f8f.MP4

**Comment by AriaYu927** [comment]
Hi @LeeSingChanYau , I reviewed your PR and found when I open the edit bar of the main folder then open and close it, the edit bar is still there. I'm not sure if it should disappear.

https://user-images.githubusercontent.com/94303659/225655117-3867e071-6053-4960-bfd1-7b50c6a49e16.mp4

**Comment by PrabhjotSembhi** [comment]
Uploading HGN APP - Google Chrome 2023-03-16 23-57-34.mp4…

First time its worked fine. But sometimes to reopen a folder for the second time you have to click twice. 
check my video at 40 seconds and 53 seconds. watch full for better understanding.

**Comment by LeeSingChanYau** [comment]
> Hi @LeeSingChanYau, I reviewed your PR and it is looking good. Good job!
> 
> I just noticed that when I open a subfolder and open the Edit, and close the first folder, the Edit button doesn't close.
> 
> So, if I open the first folder's button, both of them remain open. Print below how it looks.
> 
> image

@NataliaCavicchioli The parent edit button shouldn't close when you open or close the folder. The problem that this PR is fixing is when you open the parent folder, and click the edit button of the subtask and the options will still appear after you close the parent folder. So based on what you told me, it is behaving correctly.

**Comment by LeeSingChanYau** [comment]
@AriaYu927 Yeah I meant to leave it like that. I think it should not close the edit options after opening and closing a parent folder.

**Comment by NyahJ** [comment]
Hi @LeeSingChanYau,

Thank you for your hard work on this PR, followed your test steps: I opened a folder, then opened a subfolder that had subtasks or subfolders. Then, I closed the first folder opened, and then reopen. The subfolder closes, however, **the task under subfolder did not close**
Then I tried the "Edit" button with the same folder: I clicked the Edit button for one of subtask and the subfolder: **it worked for subfolder but not for subtask.**
In a nutshell, the subfolder is under control by its mother folder, but the subtasks are out of control of subfolders. I uploaded a vidoe for your reference. Thanks!

https://user-images.githubusercontent.com/77643813/226072706-2566baae-a8cb-428d-938c-e2fb055c7def.mp4

**Review by nicollext — APPROVED** [review]
1. **When having a sub task folder open, close and reopen the mother task’s folder. The sub task folder will remain open but not showing any sub tasks.** It is working as expected.

2. **When having a sub task edit bar open, close the mother task’s folder. The edit bar for that subtask won’t disappear**  This feature is only functioning properly when accessing a direct child of the parent folder. However, I suspect that there may be another issue causing this problem that is not within the scope of this PR. If you navigate to a subtask located within a subfolder in the Projects components and then close the main folder, the subtask remains visible on the screen. To address this issue, I have created a new item in the Bugs and Needed Functionalities Docs.

**Comment by LeeSingChanYau** [comment]
> Hi @LeeSingChanYau,
> 
> Thank you for your hard work on this PR, followed your test steps: I opened a folder, then opened a subfolder that had subtasks or subfolders. Then, I closed the first folder opened, and then reopen. The subfolder closes, however, **the task under subfolder did not close** Then I tried the "Edit" button with the same folder: I clicked the Edit button for one of subtask and the subfolder: **it worked for subfolder but not for subtask.** **In a nutshell, the subfolder is under control by its mother folder, but the subtasks are out of control of subfolders.** I uploaded a vidoe for your reference. Thanks!

Hello @NyahJ, thank you for the feedback. On my end the contents of the **Update Marketing Videos** folder are different and I can't replicate the problem you are showing me. Here is a video for reference.

https://user-images.githubusercontent.com/67836976/226084492-cbfb88bf-30e4-4144-ab5c-9bc7f6e8def8.mov

**Comment by calexander3108** [comment]
Hey @LeeSingChanYau, I tested the PR and I can reproduce the problem @NyahJ mentioned. It seems that other than the 4th folder it's producing that bug. I showed this in the video where I tried opening subfolder of folder 1 and folder 3. Seems to me that for these buggy folders(1 & 3), only the direct-parent (example: 1.1) of that folder (1.1.1) has control. Closing the main-parent (1) won't close folder (1.1.1.).

https://user-images.githubusercontent.com/127106980/226090722-d124b910-d40f-4c65-b97e-8ab9cebd9f73.mp4

File: src/components/Projects/WBS/WBSDetail/Task/Task.jsx [hunk 1]
```diff
@@ -56,6 +56,13 @@ const Task = props => {
   }, []);
   let passCurrentNum = false;
 
+  useEffect(() => {
+    console.log('isOpen', isOpen);
```
Comment (calexander3108) [inline src/components/Projects/WBS/WBSDetail/Task/Task.jsx:60]: Does this mean to get cleaned up later?

File: src/components/Projects/WBS/WBSDetail/WBSTasks.jsx [hunk 2]
```diff
@@ -201,6 +204,10 @@ const WBSTasks = props => {
   });
   const filteredTasks = filterTasks(LoadTasks, filterState);
 
+  useEffect(() => {
```
Comment (calexander3108) [inline src/components/Projects/WBS/WBSDetail/WBSTasks.jsx:207]: same cleanup questions/reminder for this useEffect() as well, seems that its only used for logging

**Comment by Jorge-Nerd** [comment]
Hi @LeeSingChanYau I tested the PR and realized a little bug when I close the task the edit still appears
image

**Comment by mpavithr** [comment]
Hi when testing, it seems to work for me, I don't see the edit furthur.
pr710_1
pr710_2

**Comment by one-community** [comment]
Would like more reviews on this, but moving to final review too because Alan is wrapping up.

**Comment by EvianTan** [comment]
The bug has already been fixed by #765 so closed this one.
